client/src/Admin/SitesForm.js

```diff
@@ -67,10 +67,17 @@ function SitesForm() {
     setData(newData);
   }
 
-  function updatePartner(index) {
-    let dropdownMenuButton = document.getElementById('dropdownMenuButton');
-    dropdownMenuButton.innerHTML = partners[index].name;
-    setSelectedPartner(partners[index]);
+  function updateAssociation(event) {
+    const newData = { ...data };
+    if (event.target.checked) {
+      newData[event.target.name].push(parseInt(event.target.value));
+    } else {
+      const index = newData[event.target.name].indexOf(parseInt(event.target.value));
+      if (index >= 0) {
+        newData[event.target.name].splice(index, 1);
+      }
+    }
+    setData(newData);
```

Comment:
This function has been written to be generic that could be used for any many-to-many association set up in the same way.

It basically checks if the box was checked or not, and adds/removes the id from the corresponding array attribute. Note that it needs to convert the value from a string to int, as input boxes end up converting everything, including the integer ids, into strings.